From 3f1bc1c541796bfbaffce9c682c8978cfe4844fd Mon Sep 17 00:00:00 2001
From: Neal Gompa <neal@gompa.dev>
Date: Sun, 27 Sep 2026 11:26:10 -0400
Subject: [PATCH] Revert "arm64: Limit memory allocations to 4GB boundary"

This causes failed memory allocations on some ARM-based platforms,
notably Apple Silicon Macs.

This reverts commit 1cae861314f43540cde773fab8b6ed769dc5f106.

Signed-off-by: Neal Gompa <neal@gompa.dev>
---
 include/grub/arm64/efi/memory.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/grub/arm64/efi/memory.h b/include/grub/arm64/efi/memory.h
index 2c64918e3..c6cb32417 100644
--- a/include/grub/arm64/efi/memory.h
+++ b/include/grub/arm64/efi/memory.h
@@ -1,6 +1,6 @@
 #ifndef GRUB_MEMORY_CPU_HEADER
 #include <grub/efi/memory.h>
 
-#define GRUB_EFI_MAX_USABLE_ADDRESS 0xffffffff
+#define GRUB_EFI_MAX_USABLE_ADDRESS 0xffffffffffffULL
 
 #endif /* ! GRUB_MEMORY_CPU_HEADER */
-- 
GitLab

